perun-core/src/main/java/cz/metacentrum/perun/core/api/MembersManager.java

```diff
@@ -36,9 +36,11 @@
 	 * @throws MemberNotExistsException
 	 * @throws PrivilegeException
 	 * @throws MemberAlreadyRemovedException
-	 * @throws GroupOperationsException
+	 * @throws WrongAttributeValueException
+	 * @throws AlreadyMemberException
+	 * @throws WrongReferenceAttributeValueException
 	 */
-	void deleteMember(PerunSession sess, Member member) throws InternalErrorException, MemberNotExistsException, PrivilegeException, MemberAlreadyRemovedException, GroupOperationsException;
+	void deleteMember(PerunSession sess, Member member) throws InternalErrorException, MemberNotExistsException, PrivilegeException, MemberAlreadyRemovedException, WrongAttributeValueException, AlreadyMemberException, WrongReferenceAttributeValueException;
```

Comment:
Shouldn't throw ``AlreadyMemberException``, gets down to ``removeDirectMember()`` in GroupsManagerBlImpl, which doesn't actually throws it inside, just method definition says it, please remove it.

Reply:
It doesn't stop in `removeDirectMember` it gets more down to `removeRelationMembers` which calls `removeIndirectMembers()`, that is the source of that exception. Does` AlreadyMemberException` make sense in `removeIndirectMembers()` method? If so, i can't delete it.